apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/PgNumericTests.cs

```diff
@@ -0,0 +1,514 @@
+        [InlineData("-10.0", "0.00000009000000000")]
+        [InlineData("0", "1.00000000")]
+        [InlineData("1", "1.01")]
+        [InlineData("0.00000009", "0.0000008")]
+        [InlineData("1.999999999999999999999999", "2")]
+        public void CompareToLessThanTest(string input1, string input2)
+        {
+            var number1 = PgNumeric.Parse(input1);
+            var number2 = PgNumeric.Parse(input2);
+            var result = number1.CompareTo(number2);
+            Assert.True(result < 0);
+        }
+
+        [Theory]
+        [InlineData("NaN", "99999999999999999999999999999")]
+        [InlineData("99.99", "99.90")]
+        [InlineData("1.0", "-1.000")]
+        [InlineData("-1", "-123.0")]
+        [InlineData("0", "-0.0009")]
+        [InlineData("0.00009", "0.00000009")]
+        public void CompareToGreaterThanTest(string input1, string input2)
+        {
+            var number1 = PgNumeric.Parse(input1);
+            var number2 = PgNumeric.Parse(input2);
+            var result = number1.CompareTo(number2);
+            Assert.True(result > 0);
+        }
+
+        [Theory]
+        [InlineData("1.00", "1")]
+        [InlineData("1", "1.00")]
+        [InlineData("NaN", "NaN")]
+        [InlineData("1.0000", "1.000")]
+        [InlineData("123456789123456789123456789123456789123456789123.123456789123456789123456789123456789123", "123456789123456789123456789123456789123456789123.123456789123456789123456789123456789123")]
+        [InlineData("-123456789123456789123456789123456789123456789123.123456789123456789123456789123456789123", "-123456789123456789123456789123456789123456789123.123456789123456789123456789123456789123")]
+        [InlineData("1.00", "1.0")]
+        [InlineData("-1.00", "-1.0")]
+        [InlineData("-0.00", "0.0")]
+        public void GetHashCodeTest(string input1, string input2)
+        {
+            var number1 = PgNumeric.Parse(input1);
+            var number2 = PgNumeric.Parse(input2);
+            var hashCode1 = number1.GetHashCode();
+            var hashCode2 = number2.GetHashCode();
+            Assert.Equal(hashCode1, hashCode2);
+        }
+
+        [Theory]
+        [InlineData("1.00")]
+        [InlineData("1.0")]
+        [InlineData("NaN")]
+        [InlineData("1.0000")]
+        [InlineData("-1.00")]
+        [InlineData("-0.00")]
+        [InlineData("0.0000000000000009")]
+        [InlineData("-0.0000000000000009")]
+        [InlineData("1234567891234567891234567891234567891234567891234567891234567891234567891234567891234567891230.12345678912345678912345678912345678912345678")]
+        [InlineData("-1234567891234567891234567891234567891234567891234567891234567891234567891234567891234567891230.12345678912345678912345678912345678912345678")]
+        public void ParseValidTest(string input)
+        {
+            var number = PgNumeric.Parse(input);
+            Assert.IsType<PgNumeric>(number);
+            Assert.NotNull(number.ToString());
```

Comment:
We don't need asserts here, Parse will throw if it cannot parse the string. In particular:

- If the first Assert executes it will always pass as PgNumeric is a struct.
- If the second Assert executes it will always pass as we never return null in ToString.

That is, you can remove the asserts.